### PR TITLE
first

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

File: src/main/java/cinema/dao/impl/TicketDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            transaction = session.beginTransaction();
+            session.save(ticket);
+            transaction.commit();
+            return ticket;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert ticket Entity with id "
+                    + ticket, e);
```
Comment (KostiantynVorobiov) [inline src/main/java/cinema/dao/impl/TicketDaoImpl.java]:
```suggestion
                    + ticket.getId, e);
```

File: src/main/java/cinema/Main.java [hunk 2]
```diff
@@ -55,5 +58,16 @@ public static void main(String[] args) {
...
         }
+        User user1 = new User();
+        user1.setEmail("sjfakljsflaks");
+        user1.setPassword("qwerty123");
+        UserService userService = (UserService) injector
+                .getInstance(UserService.class);
+        userService.add(user1);
+        ShoppingCartService shoppingCartService = (ShoppingCartService) injector
+                .getInstance(ShoppingCartService.class);
+        shoppingCartService.registerNewShoppingCart(user1);
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/Main.java]: let's add shopping cart registration to authenticationService.register()

**Review by KostiantynVorobiov — APPROVED** [review]
Good!

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> getByUserQuery = session.createQuery("FROM ShoppingCart "
+                    + "WHERE user.id = :user_id", ShoppingCart.class);
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java]: will you have access to user of this shoppingCart after session has been closed?

File: src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCartDao.getByUser(shoppingCart.getUser()).setTickets(null);
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java]: will these changes appear in DB?

File: src/main/java/cinema/Main.java (hunk 2 above)
Comment (voloshynamaryna11) [reply src/main/java/cinema/Main.java]: done

File: src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (voloshynamaryna11) [reply src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java]: done

File: src/main/java/cinema/dao/impl/TicketDaoImpl.java (hunk 1 above)
Comment (voloshynamaryna11) [reply src/main/java/cinema/dao/impl/TicketDaoImpl.java]: done

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 3 above)
Comment (voloshynamaryna11) [reply src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java]: done

File: src/main/java/cinema/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+
+@Entity
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/model/ShoppingCart.java]: let's not use fetch type Eager for OneToMany relationships. And you are using join fetch type in you query anywat

Comment (dmytro-hryhoruk) [reply src/main/java/cinema/model/ShoppingCart.java]: Why CascadeType.All?

File: src/main/java/cinema/model/ShoppingCart.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    private List<Ticket> tickets;
+    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/model/ShoppingCart.java]: What is the default fetchType for OneToOne?

Comment (voloshynamaryna11) [reply src/main/java/cinema/model/ShoppingCart.java]: done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**